src/ase/cw/model/Server.java

```diff
@@ -10,22 +13,27 @@
  */
 public class Server implements OrderConsumer {
 
+    private enum status {FREE, BUSY, PAUSED}
+
     private final BlockingQueue<Order> orderQueue;
     private final OrderHandler orderHandler;
     private int processTime = 1000;
     private Thread serverThread;
     private String name = "Server";
-    private boolean busy;
     private int serverId;
     private boolean stopThread = false;
+    private status serverStatus;
+    private ServerStatusListener ssl;
 
-    public Server(BlockingQueue<Order> queue, OrderHandler orderHandler, int serverId) {
+    public Server(BlockingQueue<Order> queue, OrderHandler orderHandler, ServerStatusListener ssl, int serverId) {
```

Comment:
Just saying:
How do you prevent that i can simply call this? 
s1 = new Server(queue,handler, 1);
s2 =new Server(queue,handler, 1);
now, two servers have the same ID, but a ID should be ID  unique, should not?

Reply:
This is a good point, I will have a look into this next